code/parse/sexp.cpp

```diff
@@ -3356,22 +3363,45 @@ int check_sexp_syntax(int node, int return_type, int recursive, int *bad_node, s
 					} else {
 						z = op_node;
 
-						if (get_operator_const(op_node) == OP_AI_DOCK)	// ai-dock with dockee
-							ship_node = CDR(z);
-						else if (type == OPF_DOCKER_POINT)
+						if (get_operator_const(op_node) == OP_AI_DOCK) { // ai-dock with dockee
```

Comment:
Same comments apply to this instance and the next one.  In fact, since the same block of code is repeated three times, it would be good to put it in a function.  A 3x repetition is a common rule of thumb for this, especially if the repeated code block is more than a couple lines long.

Reply:
No

Reply:
Well, the motivation for this rule of thumb is to make the coder's life easier.  But if you like doing 3x the amount of maintenance work, I won't stand in your way.